snuba/consumers/consumer_builder.py

```diff
@@ -138,26 +142,34 @@ def __build_consumer(self, worker: ConsumerWorker) -> StreamProcessor[KafkaPaylo
         return StreamProcessor(
             consumer,
             self.raw_topic,
-            BatchProcessingStrategyFactory(
-                worker=worker,
-                max_batch_size=self.max_batch_size,
-                max_batch_time=self.max_batch_time_ms,
-                metrics=self.metrics,
-            ),
+            strategy_factory,
```

Comment:
So this is changing a lot of how the consumer works. 
What about doing a sanity check on the errors consumer first ? It consumes the same events of the events and creates the same replacements, but none is querying the table, so you could verify that nothing obvious breaks almost for free, you just need to preserve the old initialization code and switch based on the storage (which is an initialization parameter). 
In alternative you could run this PR on a single error consumer for one hour before merging (I did that for several changes to the cdc consumer). 
What do you think ?